src/ILCompiler.Compiler/src/Compiler/CompilationBuilder.cs

```diff
@@ -92,14 +94,18 @@ public CompilationBuilder UseDebugInfoProvider(DebugInformationProvider provider
 
         public abstract CompilationBuilder UseBackendOptions(IEnumerable<string> options);
 
+        public abstract CompilationBuilder UseILProvider(ILProvider ilProvider);
```

Comment:
I do not see UseILProvider called anywhere.

Reply:
This is a future extension point. I just wanted to see how it looked like (it got a bit tricky because I didn't want `ILScanner` to also have to choose the suitable `ILProvider`, but the base `CompilationBuilder` doesn't know it either). I can delete it for now, but I do have plans to use it.